src/ds/rb_map.h

```diff
@@ -300,15 +300,15 @@ namespace rb
       return map.get_serialized_size();
     }
 
-    CBuffer& get_serialized_buffer()
+    std::span<const uint8_t>& get_serialized_buffer()
     {
       return serialized_buffer;
     }
```

Comment:
I'm confused by this API, both here and in `champ_map.h`, but it looks like they're unused. Can you remove `get_serialized_buffer()` and the `serialized_buffer` field from both?

And then, although unrelated to this PR, in a "while we're here": it seems we're relying on the fact that the `data` argument passed to `serialize()` has a (non-visible) size at least as large as `get_serialized_size()`. That's weird and dangerous. `serialize` should take a `std::span<uint8_t>` to write into, confirm that the size is large enough before it does anything, and return the size written. And in practice, it looks like all the callers would be happy with `std::vector<uint8_t> serialize()`, so we should provide that method instead/as well.

Reply:
Sure, I'll rip out the stuff that's not needed. This was done quite mechanically, so I'd missed it, thanks for pointing it out.

Agreed about this weirdness, will fix separately.